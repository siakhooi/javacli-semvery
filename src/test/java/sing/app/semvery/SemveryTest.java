package sing.app.semvery;

import static com.github.stefanbirkner.systemlambda.SystemLambda.tapSystemOut;
import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.ValueSource;
import au.com.origin.snapshots.Expect;
import au.com.origin.snapshots.junit5.SnapshotExtension;

@ExtendWith({SnapshotExtension.class})
class SemveryTest {
    private Expect expect;

    @Test
    void callMainWithNoArguments() throws Exception {
        String text = tapSystemOut(() -> Semvery.main(new String[] {}));
        assertDoesNotThrow(() -> expect.toMatchSnapshot(text));
    }

    @Test
    void callMainWithArguments() throws Exception {
        String text = tapSystemOut(() -> Semvery.main(new String[] {"ABC", "CDE"}));
        assertDoesNotThrow(() -> expect.toMatchSnapshot(text));
    }

    @ParameterizedTest
    @ValueSource(strings = {"-h", "--help"})
    void callMainWithHelp(String argument1) throws Exception {
        String text = tapSystemOut(() -> Semvery.main(new String[] {argument1}));
        assertDoesNotThrow(() -> expect.toMatchSnapshot(text));
    }

    @ParameterizedTest
    @ValueSource(strings = {"-v", "--version"})
    void callMainWithVersion(String argument1) throws Exception {
        String text = tapSystemOut(() -> Semvery.main(new String[] {argument1}));
        assertDoesNotThrow(() -> expect.toMatchSnapshot(text));
    }
}
