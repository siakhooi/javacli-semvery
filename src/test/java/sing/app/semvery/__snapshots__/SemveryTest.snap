sing.app.semvery.SemveryTest.callMainWithArguments=[
semvery 0.6.0
Usage: semvery [options]
  Options:
    --operation, -o
      Options: [isValid, isStable]
    --any
      Return success(status 0) if any of the values meet the operation 
      criteria, instead of all values have to meet.
      Default: false
    --version, -v
      Display version
    --help, -h
      Display help

Details:
  -o isValid version [version...]
  --operation isValid version [version...]

    Example:
      -o isValid 1.0.0
      -o isValid ABC
      -o isValid 1.0.0 2.0.0
      -o isValid 1.0.0 ABC

  -o isStable version [version...]
  --operation isStable version [version...]

    Example:
      -o isStable 1.0.0
      -o isStable 0.1.0
      -o isStable 1.0.0 0.1.0

]


sing.app.semvery.SemveryTest.callMainWithHelp=[
semvery 0.6.0
Usage: semvery [options]
  Options:
    --operation, -o
      Options: [isValid, isStable]
    --any
      Return success(status 0) if any of the values meet the operation 
      criteria, instead of all values have to meet.
      Default: false
    --version, -v
      Display version
    --help, -h
      Display help

Details:
  -o isValid version [version...]
  --operation isValid version [version...]

    Example:
      -o isValid 1.0.0
      -o isValid ABC
      -o isValid 1.0.0 2.0.0
      -o isValid 1.0.0 ABC

  -o isStable version [version...]
  --operation isStable version [version...]

    Example:
      -o isStable 1.0.0
      -o isStable 0.1.0
      -o isStable 1.0.0 0.1.0

]


sing.app.semvery.SemveryTest.callMainWithNoArguments=[
semvery 0.6.0
Usage: semvery [options]
  Options:
    --operation, -o
      Options: [isValid, isStable]
    --any
      Return success(status 0) if any of the values meet the operation 
      criteria, instead of all values have to meet.
      Default: false
    --version, -v
      Display version
    --help, -h
      Display help

Details:
  -o isValid version [version...]
  --operation isValid version [version...]

    Example:
      -o isValid 1.0.0
      -o isValid ABC
      -o isValid 1.0.0 2.0.0
      -o isValid 1.0.0 ABC

  -o isStable version [version...]
  --operation isStable version [version...]

    Example:
      -o isStable 1.0.0
      -o isStable 0.1.0
      -o isStable 1.0.0 0.1.0

]


sing.app.semvery.SemveryTest.callMainWithVersion=[
0.6.0

]